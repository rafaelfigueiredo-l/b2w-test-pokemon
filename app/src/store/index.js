import { createStore } from "redux";

const INITIAL_STATE = {
  theme: {},
  pokemon: [],
};
function course(state = INITIAL_STATE, action) {
  switch (action.type) {
    case "SET_THEME":
      return {
        ...state,
        theme: action.payload,
      };

    case "SET_POKEMON":
      return {
        ...state,
        pokemon: action.payload,
      };

    default:
      return state;
  }
}

const store = createStore(
  course,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
