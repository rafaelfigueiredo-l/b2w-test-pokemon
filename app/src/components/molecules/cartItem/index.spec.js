import { Provider } from "react-redux";
import store from "../../../store";
import CartItem from "./";

describe("<CartItem />", () => {
  const component = (
    <Provider store={store}>
      <CartItem />
    </Provider>
  );

  const mounted = mount(component);

  it("should mount correctly CartItem component", () => {
    expect(mounted.find(".cartItem")).toHaveLength(2);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly CartItem component", () => {
    expect(mounted.unmount()).toEqual({});
  });
});
