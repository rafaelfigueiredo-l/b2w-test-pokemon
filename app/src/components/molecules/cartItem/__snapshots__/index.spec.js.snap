// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartItem /> should mount correctly CartItem component 1`] = `
<styled.div
  className="cartItem row"
>
  <div
    className="sc-AxgMl gmXeLJ cartItem row"
  >
    <styled.div
      className="column"
    >
      <div
        className="sc-AxheI jLIeQh column"
      >
        <span
          className="image"
        >
          <styled.img
            alt="Loader"
            className="loader"
            src="loader.gif"
            width="50px"
          >
            <img
              alt="Loader"
              className="sc-AxirZ gEUYej loader"
              src="loader.gif"
              width="50px"
            />
          </styled.img>
        </span>
      </div>
    </styled.div>
    <styled.div
      className="column"
    >
      <div
        className="sc-AxheI jLIeQh column"
      >
        <styled.span
          className="cartItemName"
        >
          <span
            className="sc-AxhCb ccMvAG cartItemName"
          />
        </styled.span>
        <styled.span
          className="cartItemPrice"
        >
          <span
            className="sc-AxhUy kMyssT cartItemPrice"
          />
        </styled.span>
      </div>
    </styled.div>
    <styled.div
      className="divider"
    >
      <div
        className="sc-AxjAm jjIFiv divider"
      />
    </styled.div>
  </div>
</styled.div>
`;
